logstash-core/spec/logstash/instrument/wrapped_write_client_spec.rb

```diff
@@ -18,39 +18,54 @@
 
   subject { described_class.new(write_client, pipeline, metric, plugin) }
 
+  def threaded_read_client(read_client)
+    Thread.new(read_client) do |_read_client|
```

Comment:
strangely, I can't make that copy happen:

```ruby
require 'rspec'

RSpec.describe Thread do
  context "variable copying in Thread.new" do
    let(:hash) { Hash.new }
    context "when variable is not passed as an argument" do
      it "does not copy the value" do
        t = Thread.new { Thread.current[:var] = hash }.join
        expect(t[:var]).to be(hash)
      end
    end
    context "when variable is passed as an argument" do
      it "copies the value" do
        t = Thread.new(hash) {|h| Thread.current[:var] = h }.join
        expect(t[:var]).to_not be(hash)
      end
    end
  end
end
```

and running it:

```
% rspec thread_spec.rb
.F

Failures:

  1) Thread variable copying in Thread.new when variable is passed as an argument copies the value
     Failure/Error: expect(t[:var]).to_not be(hash)
     
       expected not #<Hash:2050> => {}
                got #<Hash:2050> => {}
     
       Compared using equal?, which compares object identity.
     
     
       Diff:
       
     # ./thread_spec.rb:15:in `(root)'

Finished in 0.418 seconds (files took 0.995 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./thread_spec.rb:13 # Thread variable copying in Thread.new when variable is passed as an argument copies the value
```

in both situations the value isn't copied, so the let is only executed once